articles/active-directory/develop/quickstarts/active-directory-ios.md

```diff
@@ -0,0 +1,180 @@
+
+    ```xml
+    <key>CFBundleURLTypes</key>
+    <array>
+        <dict>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>CFBundleURLName</key>
+            <string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
+            <key>CFBundleURLSchemes</key>
+            <array>
+                <string>msal[Your_Application_Id_Here]</string>
+                <string>auth</string>
+            </array>
+        </dict>
+    </array>
+    ```
+
+## More Information
+
+Read these sections to learn more about this quickstart.
+
+### MSAL
+
+MSAL ([MSAL.framework](https://github.com/AzureAD/microsoft-authentication-library-for-objc)) is the library used to sign in users and request tokens used to access an API protected by Microsoft Azure Active Directory. You can use carthage to install it by first running the following command in bash terminal from App’s root folder to build MSAL:
+
+```bash
+echo "github \"AzureAD/microsoft-authentication-library-for-objc\" \"master\"" > Cartfile
+carthage update
+```
+
+Then, in XCode, open the **General**, go to **Linked Frameworks and Libraries** and add **MSAL.framework** in **Add other...**, then add the following content in **New Run Script Phase**:
+
+```text
+/usr/local/bin/carthage copy-frameworks
+```
+
+And the following to **Input Files**:
+
+```text
+$(SRCROOT)/Carthage/Build/iOS/MSAL.framework
+```
+
+### MSAL initialization
+
+You can add the reference for MSAL by adding the following code:
+
+```swift
+import MSAL
+```
+
+Then, initialize MSAL using the following code:
+
+```swift
+self.applicationContext = try MSALPublicClientApplication.init(clientId: kClientID, authority: kAuthority)
+```
+
+> |Where: ||
+> |---------|---------|
+> | `clientId` | The Application ID from the application registered in *portal.microsoft.com* |
+> | `authority` | The Azure AD v2.0 endpoint. In most of cases this will be *https<span/>://login.microsoftonline.com/common/v2.0* |
+
+### Requesting tokens
+
+MSAL has two methods used acquire tokens: `acquireToken` and `acquireTokenSilent`.
+
+#### Getting a user token interactively
+
+Some situations require forcing users to interact with Azure Active Directory (Azure AD) v2.0 endpoint which will result in a context switch to the system browser to either validate users's credentials or for consent. Some examples include:
+
+* The first time users sign in to the application
+* When users may need to reenter their credentials because the password has expired
+* When your application is requesting access to a resource that the user needs to consent to
+* When two factor authentication is required
+
+```swift
+self.applicationContext.acquireToken(forScopes: self.kScopes)
+```
+
+> |Where:||
+> |---------|---------|
+> | `forScopes` | Contains the scopes being requested (that is, `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs) |
```

Comment:
Since input scopes is an array of strings, let's use an array in this sample.

So:

`[ "user.read" ]` and `[ "api://<Application ID>/access_as_user" ]`